client/pool.c

```diff
@@ -815,17 +817,20 @@ void *pool_main_thread(void *arg)
 		{
 			LL_DELETE(g_accept_connection_list_head, elt);
 			LL_APPEND(g_connection_list_head, elt);
+			g_connection_changed = 1;
 		}
-		
 		int index = 0;
-		LL_FOREACH(g_connection_list_head, elt)
-		{
-			memcpy(g_fds + index, &elt->connection_data.connection_descriptor, sizeof(struct pollfd));
-			++index;
+		if(g_connection_changed) {
+			g_connection_changed = 0;
+			LL_FOREACH(g_connection_list_head, elt)
+			{
+				memcpy(g_fds + index, &elt->connection_data.connection_descriptor, sizeof(struct pollfd));
+				++index;
+			}
 		}
-		pthread_mutex_unlock(&g_descriptors_mutex);
 		
-		int res = poll(g_fds, index, 1000);
+		int res = poll(g_fds, g_connections_count, 1000);
+		pthread_mutex_unlock(&g_descriptors_mutex);
```

Comment:
I don't think this is a good change. g_fds is accessed only from one thread, so it does not require synchronization. poll function can block execution for some time, so g_descriptors_mutex can be blocked for the long time.